tests/mir-opt/copy-prop/mutate_through_pointer.rs

```diff
@@ -8,7 +8,7 @@ fn f(c: bool) -> bool {
     mir!({
         let a = c;
         let p = core::ptr::addr_of!(a);
-        let p2 = core::ptr::addr_of_mut!(*p);
+        let p2 = core::ptr::addr_of_mut!(a);
         *p2 = false;
```

Comment:
> This attempts to mutate `a` via a pointer derived from `addr_of!(a)`. That is UB according to Miri. However, the decision to make this UB - and to allow rustc to rely on that fact for the purpose of optimizations - has not been finalized.
>
> As such, we include this test to ensure that copy prop does not rely on that fact. Specifically, we ensure that we do not propagate `a` into `c`, which would cause `f(true)` to incorrectly return `false`.